image-flash.c

```diff
@@ -50,8 +50,15 @@ static int flash_generate(struct image *image)
 		}
 		mode = MODE_APPEND;
 
-		if (!part->image)
+		if (!part->image) {
+			ret = pad_file(image, NULL, part->size, 0xFF, mode);
+			if (ret) {
+				image_error(image, "failed to pad partition '%s'\n",
+						part->name);
+				return ret;
+			}
 			continue;
+		}
```

Comment:
This is mostly duplicated code. Maybe like this:
```
if (part->image) {
        ...
        infile = imageoutfile(child);
} else
        infile = NULL;

ret = pad_file(image, infile, part->size, 0xFF, mode);
...
```

Reply:
True. Since #101 was merged I have also dropped the child check
```
			if (!child) {
				image_error(image, "could not find %s\n", part->name);
				return -EINVAL;
			}
```
because that is now handled by `flash_setup`

Reply:
Turns out we can't. See #104